frontend/src/container/AdvancedSearch.jsx

```diff
@@ -0,0 +1,303 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { translate } from 'react-i18next'
+import {
+  PageWrapper,
+  PageTitle,
+  PageContent,
+  ListItemWrapper,
+  displayDistanceDate,
+  IconButton,
+  BREADCRUMBS_TYPE,
+  CUSTOM_EVENT,
+  buildHeadTitle,
+  PAGE,
+  TracimComponent
+} from 'tracim_frontend_lib'
+import ContentItemSearch from '../component/Search/ContentItemSearch.jsx'
+import ContentItemHeader from '../component/Workspace/ContentItemHeader.jsx'
+import {
+  newFlashMessage,
+  setCurrentNumberPage,
+  appendSearchResultsList,
+  setSearchResultsList,
+  setNumberResultsByPage,
+  setSearchedKeywords,
+  setBreadcrumbs,
+  setHeadTitle
+} from '../action-creator.sync.js'
+import { getSearchedKeywords } from '../action-creator.async.js'
+import Search from '../component/Search/Search.jsx'
+import { parseSearchUrl } from '../util/helper.js'
+import SearchFilterMenu from '../component/Search/SearchFilterMenu.jsx'
+import classnames from 'classnames'
+
+const qs = require('query-string')
+
+export class AdvancedSearch extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      totalHits: 0,
+      isFilterMenuOpen: false
+    }
+
+    props.registerCustomEventHandlerList([
+      { name: CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, handler: this.handleAllAppChangeLanguage }
+    ])
+  }
+
+  handleAllAppChangeLanguage = () => {
+    this.setHeadTitle()
+    this.buildBreadcrumbs()
+  }
+
+  componentDidMount () {
+    this.setHeadTitle()
+    this.buildBreadcrumbs()
+    this.loadSearchUrl()
+  }
+
+  componentDidUpdate (prevProps) {
+    const { props } = this
+    const prevSearchedKeywords = qs.parse(prevProps.location.search).q
+    const currentSearchedKeywords = parseSearchUrl(qs.parse(props.location.search)).searchedKeywords
```

Comment:
why isn't it the same function call in these 2 lines ?
I would assume calling the same function with only the parameter changing from prevProps to props

especially because parseSearchUrl convert the q query param to searchedKeywords

Reply:
No specific reason, I will fix that.